libsentrykube/kube.py

```diff
@@ -111,6 +112,13 @@ def _consolidate_variables(
        overrides are managed manually and they can contain comments. Tools cannot
        preserve comments.
     4. overridden by the cluster file. Which is likely going to be replaced by 2 and 3.
+    5. overridden by creating a hierarchical structure. Adding an intermediate directory
+       in 'region_override' with a '_values.yaml' file allows to have a common config
+       between a set of regions. This is useful if regions in a service are different, but
+       subset of them are similar.
+       It works like point 1 and 2 but with an additional layer in between
+    6. overridden by a common '_values.yaml' within a region folder that has a shared config
+       between all clusters in the region. Values in point 2 will still override those values
```

Comment:
I don't think the order is right.

1. Cluster file will be deprecated for service specific override, we are moving to region override only. It still exists because we are in the middle of this process. But if a service has region overrides there should be nothing in the cluster file for that service.
2. managed override (point 3 above) has to be the last override file (assuming the cluster file goes away). 
3. It seems that the common `_values.yaml` file within a region (point 6) should be before point 2 (region override).  If it is shaerd between clusetrs in the same region, the cluster specific file (these - https://github.com/getsentry/ops/tree/master/k8s/services/relay-pop/region_overrides/us) should be more specific than you point 6. 
4. your point 5 should be before 2 as well. 

In summary This should be the order from more generic to more specific
1 The per service value file (this is not region or cluster specific. It is service specific)
5 common override between subset of regions of the same service 
6 common override between clusters in the same region (more specific than 5)
3 cluster specific override 
4 managed override, which is basically equivalent to 3 but managed by tools. 

4 the cluster file is still there for historic reasons, but if you made changes to any of the above the cluster file is supposed to be empty for the specific service.

Reply:
Thanks for the detailed comment. You're right that the order isn't correct, I misunderstood the existing comment and thought it's more of a "we have this kind of overrides" instead of them being the order from most to least specific. I will update it accordingly